contracts/InterchainTokenService.sol

```diff
@@ -663,21 +658,26 @@ contract InterchainTokenService is
     \****************/
 
     function _setup(bytes calldata params) internal override {
-        (address operator, string memory chainName_, string[] memory trustedChainNames, string[] memory trustedAddresses) = abi.decode(
+        (address operator, string memory chainName_, string memory itsHubAddress_, string[] memory trustedChainNames) = abi.decode(
             params,
-            (address, string, string[], string[])
+            (address, string, string, string[])
         );
-        uint256 length = trustedChainNames.length;
-
         if (operator == address(0)) revert ZeroAddress();
         if (bytes(chainName_).length == 0 || keccak256(bytes(chainName_)) != chainNameHash) revert InvalidChainName();
-        if (length != trustedAddresses.length) revert LengthMismatch();
+        if (bytes(itsHubAddress_).length == 0) revert InvalidHubAddress();
 
         _addOperator(operator);
         _setChainName(chainName_);
+        _setItsHubAddress(itsHubAddress_);
 
+        uint256 length = trustedChainNames.length;
         for (uint256 i; i < length; ++i) {
-            _setTrustedAddress(trustedChainNames[i], trustedAddresses[i]);
+            string memory trustedChainName = trustedChainNames[i];
+            _setTrustedChain(trustedChainName);
+            // Remove previously set trusted addresses.
```

Comment:
```suggestion

            // Migration: Remove previously set trusted addresses.
```
we should add a dedicated `_migrate` method in the future that's only run when not doing the initial deployment

Reply:
I feel this would be inelegant here: what would this `_migrate` function take as input? If we want to be generic it would be the `bytes params` which would be re-decoded and the chains would be iterated over again, to do a simple check. Alternatively, migrate could take in a chain name and only do 
```
if (trustedAddressHash(trustedChainName) != 0) {
    _removeTrustedAddress(trustedChainName);
}
```
which is fine but not meaningful for the future. What do you think?

Reply:
yeah, we'll need to also change how the `Upgradable` works. Currently `setup` gets called both in the proxy constructor and during `upgrade`. However, `upgrade` might require dedicated migration code that shouldn't be run in the constructor. It's not been a major issue yet since we haven't needed to do many migrations. It's still not a major issue here so it's fine as-is. In the future, we could add a separate `migrate` method to `Upgradable` that gets called during `upgrade` whereas `setup` only gets called from the constructor, and the logic can differ then (e.g. `setup` doesn't need to remove trusted addresses if one exists)

Reply:
ok, will leave as is but keep in mind. When making this I was surprised that setup was used for both as well.